ls1intum/Artemis
# [Feature] Empty Programming Exercises

### Checklist

### Motivation and Context


Resolves #3308 

### Description

This PR adds support to create empty programming exercises. It adds the "Empty" drop-down item in the programming language list and generates empty template, test, and solution repositories. It also adds a dummy problem statement and a bare-bone build plan for Bamboo and Jenkins

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Click on Generate New Programming Exercise
4. Select "Empty" from the drop-down list of the programming language
5. Generate the exercise
6. Make sure that the template, tests, and solution repositories are created. They should also be empty.

_**Please delete the empty exercise after finishing testing it.**_

### Test Coverage



BambooBuildPlanService 95%
JenkinsBuildPlanService 99%
programming-exercise-update.component.ts 79.76%

### Screenshots


![image](https://user-images.githubusercontent.com/11298674/119970525-1d339080-bfb0-11eb-8a50-e036ac283188.png)
